go-controller/pkg/ovn/egressip.go

```diff
@@ -2288,29 +1431,41 @@ func (e *egressIPController) addPodEgressIPAssignment(egressIPName string, statu
 			metrics.RecordEgressIPAssign(duration)
 		}()
 	}
-
-	ops, err := createNATRuleOps(e.nbClient, nil, podIPs, status, egressIPName)
-	if err != nil {
-		return fmt.Errorf("unable to create NAT rule ops for status: %v, err: %v", status, err)
-	}
-
-	ops, err = e.createReroutePolicyOps(ops, podIPs, status, egressIPName)
-	if err != nil {
-		return fmt.Errorf("unable to create logical router policy ops, err: %v", err)
+	isLocalZoneNode, loadedEgressNode := e.nodeZoneState.Load(status.Node)
+	isLocalZonePod, loadedPodNode := e.nodeZoneState.Load(pod.Spec.NodeName)
+	var ops []ovsdb.Operation
+	if loadedEgressNode && isLocalZoneNode.(bool) {
+		ops, err = createNATRuleOps(e.nbClient, nil, podIPs, status, egressIPName)
+		if err != nil {
+			return fmt.Errorf("unable to create NAT rule ops for status: %v, err: %v", status, err)
+		}
+		if config.OVNKubernetesFeature.EnableInterconnect && (loadedPodNode && !isLocalZonePod.(bool)) {
+			// configure reroute for non-local-zone pods on egress nodes
```

Comment:
> i'm a little confused when we are adding static routes and policies. So if the egress node is local and the pod is not local, then we configure a static route on the pod's node?

correct.

> 
> The static route reroutes towards the GR IP of the egress node.

yes.

> 
> Then for the policy...if the pod node is local then we create a reroute policy?

yes.

> 
> The policy then:
> 
> 1. routes to GR IP if the egress node is local OR
> 2. routes to the transit router IP
> 
> Is that right? If so, could we have that in this function's comments?

yes you got all that right. See https://docs.google.com/presentation/d/1jdZrkFVTc9O9-d2I6SCa8EsBCaaxg2dD_x0868Izqdw/edit#slide=id.g229128c4b5c_0_5
I am actually doing a detailed egressIP on Interconnect documentation: https://github.com/ovn-org/ovn-kubernetes/pull/3526. This will all be outlined there. Is that ok or do you also want to see some in the code?

> 
> Is there a reason we use static routes for the reroute instead of policy? I was thinking that if we only used policy then we could use a unique id since only one should exist on a router per pod so we could never have a case where a static route and a policy were configured on the same node. We could use client side indexes to update the policy in case egress node moves and not have to worry about if we left some stale static route or policy around. wdyt?

We spoke about this in huddle, we can't use LRPs because LRSR is evaluated first in OVN pipeline and as soon as traffic is sent to the remote node, it get's dropped before even reaching LRP if we don't have the right LRSR's
As for the indexes part, I think +1 to that idea, but that is going to take considerable cycles, so as decided let's ensure to track a feature work or improvements card or something for this.